language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/contextproviders/FunctionDefinitionContextProvider.java

```diff
@@ -0,0 +1,195 @@
+//            }
+//
+//            BObjectTypeSymbol objectType = (BObjectTypeSymbol) (((SymbolInfo) filtered.get()).getScopeEntry().symbol);
+//
+//            objectType.attachedFuncs.stream()
+//                    .filter(attachedFunc -> !attachedFunc.symbol.bodyExist)
+//                    .forEach(attachedFunc -> {
+//                        String functionName = attachedFunc.funcName.getValue();
+//                        List<String> funcArguments = getFuncArguments(attachedFunc.symbol);
+//                        String label = functionName + "(" + String.join(", ", funcArguments) + ")";
+//                        if (!(attachedFunc.symbol.retType instanceof BNilType)) {
+//                            label += " returns " + generateTypeDefinition(null, objectType.pkgID,
+//                                    attachedFunc.symbol.retType);
+//                        }
+//                        String insertText = label + " {" + CommonUtil.LINE_SEPARATOR + "\t${1}"
+//                                + CommonUtil.LINE_SEPARATOR + "}";
+//                        completionItems.add(BFunctionCompletionItemBuilder.build(attachedFunc.symbol, label,
+//                                insertText));
+//                    });
+//        } else if (consumedTokens.size() == 2 && consumedTokens.get(0).equals(UtilSymbolKeys.FUNCTION_KEYWORD_KEY)
+//                && !CommonUtil.getLastItem(consumedTokens).equals(".")) {
+//            return context.get(CompletionKeys.VISIBLE_SYMBOLS_KEY).stream()
+//                    .filter(symbolInfo -> symbolInfo.getScopeEntry().symbol instanceof BObjectTypeSymbol)
+//                    .map(symbolInfo -> {
+//                        BSymbol symbol = symbolInfo.getScopeEntry().symbol;
+//                        String symbolName = symbol.getName().getValue();
+//                        CompletionItem completionItem = BTypeCompletionItemBuilder.build((BTypeSymbol) symbol,
+//                                symbolName);
+//                        completionItem.setInsertText(symbolName + ".");
+//                        return completionItem;
+//                    }).collect(Collectors.toList());
+//        }
+        return completionItems;
+    }
+
+    private List<CompletionItem> getObjectAttachedFunctions(LSContext context, List<CommonToken> lhsDefaultTokens) {
+        String objectName = lhsDefaultTokens.get(1).getText();
+        List<CompletionItem> completionItems = new ArrayList<>();
+        Optional filtered = context.get(CompletionKeys.VISIBLE_SYMBOLS_KEY)
+                .stream()
+                .filter(symbolInfo -> {
+                    BSymbol symbol = symbolInfo.getScopeEntry().symbol;
+                    return symbol instanceof BObjectTypeSymbol && symbol.getName().getValue().equals(objectName);
+                }).findFirst();
+
+        if (!(filtered.isPresent()
+                && ((SymbolInfo) filtered.get()).getScopeEntry().symbol instanceof BObjectTypeSymbol)) {
```

Comment:
Use `Optional<SymbolInfo> filtered` in [line 163](https://github.com/ballerina-platform/ballerina-lang/pull/15128/files#diff-ca442fc9d89ebcabc664b64b7b7856c4R163), then you don't need to cast it.

Reply:
Fixed